build_modules/lib/src/platform.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:build/build.dart';
+import 'package:crypto/crypto.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'platform.g.dart';
+
+final _librariesId = AssetId(r'$sdk', 'lib/libraries.json');
+final _platformsLoader = PlatformsLoader._();
+final platformsLoaderResource = Resource<PlatformsLoader>(
+    () => _platformsLoader,
+    dispose: (loader) => loader.invalidate());
+
+/// Provides the ability to safely share a single [Platforms] instance between
+/// multiple build steps.
+///
+/// A singleton instance is provided via the [platformsLoaderResource] which
+/// calls [invalidate] between builds.
+class PlatformsLoader {
+  Future<Platforms> _platforms;
+  Digest _previousDigest;
+  bool _invalidated = true;
+
+  PlatformsLoader._();
+
+  Future<Platforms> load(AssetReader reader) async {
+    if (_invalidated) {
```

Comment:
[optional - not worth a rebase if you planned on merging instead of squashing] invert the condition and bail out early to reduce 1 level of nesting?

Reply:
done